src/dataset/dataset.service.js

```diff
@@ -88,6 +88,39 @@ angular.module('vlui')
       return updatePromise;
     };
 
+    function getFieldDefs(schema, order) {
+      var fieldDefs = schema.fields().map(function(field) {
+        return {
+          field: field,
+          type: schema.type(field),
+          primitiveType: schema.primitiveType(field)
+        };
+      });
+
+      fieldDefs = util.stablesort(fieldDefs, order || Dataset.fieldOrderBy.typeThenName, Dataset.fieldOrderBy.field);
+
+      fieldDefs.push({ field: '*', aggregate: vl.aggregate.AggregateOp.COUNT, type: vl.type.QUANTITATIVE, title: 'Count' });
+      return fieldDefs;
+    }
+
+    // TODO: remove
+    Dataset.domain = function(field) {
```

Comment:
We should be able to use `schema.domain` instead (Note a difference in terms of parameter -- schema.domain takes an encodingQuery as input.)